src/libostree/ostree-repo-pull.c

```diff
@@ -1820,10 +1820,44 @@ scan_one_metadata_object (OtPullData                 *pull_data,
             return FALSE;
         }
 
+      g_autoptr(GError) local_error = NULL;
       if (!_ostree_repo_import_object (pull_data->repo, pull_data->remote_repo_local,
                                        objtype, checksum, pull_data->importflags,
-                                       cancellable, error))
-        return FALSE;
+                                       cancellable, &local_error))
+        {
+          /* When traversing parents, do not fail on a missing commit.
```

Comment:
Tempting to refactor this somehow to share with the remote case.

At least let's add a comment somewhere around here to say that this matches the remote logic in `meta_fetch_on_complete`.

Reply:
I gave that a little thought but couldn't come up with the right abstraction. `handle_missing_commit (OtPullData *, FetchObject Data *, const char*)`? Then it would queue the tombstone fetch or import it based on `remote_repo_local`.

Reply:
Added something on the last commit. I tried to figure out how to encapsulate the condition as well, but it I couldn't come up with any ideas about handling the slightly different outer logic between the 2 cases. Let me know if you see a better plan.